examples/optimistic-auction/contracts/PodAuctionConsumer.sol

```diff
@@ -0,0 +1,130 @@
+        bytes32 uniqueAuctionId = getUniqueAuctionId(auctionId, deadline);
+        State storage s = state[uniqueAuctionId];
+        require(block.timestamp >= deadline, "Writing period has not started");
+        require(block.timestamp < deadline + U, "Writing period ended");
+        require(s.winner.bid == 0, "Bid already exists");
+        require(msg.sender == bidder, "Invalid caller");
+
+        uint256 f = podRegistry.getFaultTolerance();
+        bool verified =
+            PodECDSA.verifyCertifiedLog(PodECDSA.PodConfig({quorum: 2 * f + 1, registry: podRegistry}), certifiedLog);
+        require(verified, "Invalid proof or quorum not reached");
+
+        s.winner.bidder = bidder;
+        s.winner.bid = bid;
+
+        emit BidWritten(uniqueAuctionId, bidder, bid);
+    }
+
+    function read(uint256 auctionId, uint256 deadline) external view returns (State memory) {
+        bytes32 uniqueAuctionId = getUniqueAuctionId(auctionId, deadline);
+        require(block.timestamp >= deadline + 2 * U, "Dispute period NOT ended");
+        return state[uniqueAuctionId];
+    }
+
+    function blameIllAnnounced(PodECDSA.CertifiedLog calldata certifiedLog)
+        external
+        validLog(certifiedLog.log)
+        onlyBonded
+    {
+        (uint256 auctionId, address bidder, uint256 deadline, uint256 bid) = decodeBid(certifiedLog.log);
+        bytes32 uniqueAuctionId = getUniqueAuctionId(auctionId, deadline);
+        State storage s = state[uniqueAuctionId];
+        require(s.winner.bid != 0, "Bid not written");
+        require(block.timestamp < deadline + 2 * U, "Dispute period ended");
+
+        uint256 f = podRegistry.getFaultTolerance();
+        bool verified =
+            PodECDSA.verifyCertifiedLog(PodECDSA.PodConfig({quorum: 2 * f + 1, registry: podRegistry}), certifiedLog);
+        require(verified, "Invalid proof or quorum not reached");
+
+        s.blamed.bid = s.winner.bid;
+        s.blamed.bidder = s.winner.bidder;
+
+        s.winner.bid = bid;
+        s.winner.bidder = bidder;
+
+        slash(s.blamed.bidder);
+        emit BidBlamed(uniqueAuctionId, s.blamed.bidder, s.blamed.bid);
+    }
+
+    function blameNoShow(PodECDSA.CertifiedLog calldata certifiedLog) external validLog(certifiedLog.log) onlyBonded {
+        (uint256 auctionId, address bidder, uint256 deadline, uint256 bid) = decodeBid(certifiedLog.log);
+        bytes32 uniqueAuctionId = getUniqueAuctionId(auctionId, deadline);
+        State storage s = state[uniqueAuctionId];
+
+        require(s.winner.bid == 0, "Bid already exists");
+        require(block.timestamp >= deadline + U, "Still in waiting period");
+        require(block.timestamp < deadline + 2 * U, "Dispute period ended");
+        require(s.blamed.bid < bid, "Invalid blamed bid");
+
+        uint256 f = podRegistry.getFaultTolerance();
+        bool verified =
+            PodECDSA.verifyCertifiedLog(PodECDSA.PodConfig({quorum: f + 1, registry: podRegistry}), certifiedLog);
```

Comment:
Quorum typo?

```suggestion
            PodECDSA.verifyCertifiedLog(PodECDSA.PodConfig({quorum: 2 * f + 1, registry: podRegistry}), certifiedLog);
```

Reply:
no, the auction protocol uses different quorum in different places.

Reply:
oh, I see